### PR TITLE
Update candle code

### PR DESCRIPTION
**Pull request by ghost**
No, really, I'm serious. Everything's gotta be improved right ?
- Extend description
- Fix pathing
- Use span, visible_message

### CODE REVIEW

File: code/game/objects/items/candle.dm [hunk 1]
```diff
@@ -8,80 +8,70 @@
...
-			var/obj/item/candle/C = W
-			if(C.lit)
-				light()
+/obj/item/candle/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	..()
+	if(istype(W, /obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/WT = W
+		if(WT.isOn()) //Badasses dont get blinded by lighting their candle with a welding tool
+			light("<span class='warning'>[user] casually lights [src] with [W], what a badass.</span>")
+	else if(istype(W, /obj/item/weapon/lighter))
```
Comment (ComicIronic) [inline code/game/objects/items/candle.dm]: If you really want to clean this up, there's an object proc called is_hot() which can be called to the same effect, and would have less snowflake associated with it.
You can just check if(W.is_hot()) to have the message.

Comment (dylanstrategie) [inline code/game/objects/items/candle.dm]: I'm pretty sure it'll be fine. The flavor text is nice

Comment (ComicIronic) [inline code/game/objects/items/candle.dm]: Except all the messages are the same, and snowflake code is messy. is_hot() is much neater. You could even leave the welder message in, it's just pointless to have to add in exceptions to candle code when a general system is better.

Comment (dylanstrategie) [inline code/game/objects/items/candle.dm]: I'll look into it at some point, I swear

**Pull request merged**